apps/roam/src/utils/syncDgNodesToSupabase.ts

```diff
@@ -0,0 +1,398 @@
+import {
+  getAllDiscourseNodesSince,
+  nodeTypeSince,
+} from "./getAllDiscourseNodesSince";
+import { cleanupOrphanedNodes } from "./cleanupOrphanedNodes";
+import {
+  getLoggedInClient,
+  getSupabaseContext,
+  SupabaseContext,
+} from "./supabaseContext";
+import { fetchEmbeddingsForNodes } from "./fetchEmbeddingsForNodes";
+import { LocalContentDataInput } from "@repo/database/inputTypes";
+import { RoamDiscourseNodeData } from "./getAllDiscourseNodesSince";
+import getDiscourseRelations from "./getDiscourseRelations";
+import getDiscourseNodes, { DiscourseNode } from "./getDiscourseNodes";
+import {
+  discourseNodeBlockToLocalConcept,
+  discourseNodeSchemaToLocalConcept,
+  orderConceptsByDependency,
+  discourseRelationSchemaToLocalConcept,
+  discourseRelationDataToLocalConcept,
+} from "./conceptConversion";
+import getDiscourseRelationTriples from "./getDiscourseRelationTriples";
+import { OnloadArgs } from "roamjs-components/types";
+import { DGSupabaseClient } from "@repo/ui/lib/supabase/client";
+
+const SYNC_FUNCTION = "embedding";
+const SYNC_INTERVAL = "45s";
+const SYNC_TIMEOUT = "20s";
+const BATCH_SIZE = 200;
+const DEFAULT_TIME = "1970-01-01";
+const EMBEDDING_MODEL = "openai_text_embedding_3_small_1536";
+
+type SyncTaskInfo = {
+  lastUpdateTime: string | null;
+  spaceId: number;
+  worker: string;
+  shouldProceed: boolean;
+};
+
+export const endSyncTask = async (
+  worker: string,
+  status: "complete" | "failed",
+): Promise<void> => {
+  try {
+    const supabaseClient = await getLoggedInClient();
+    const context = await getSupabaseContext();
+    if (!context) {
+      console.error("endSyncTask: Unable to obtain Supabase context.");
+      return;
+    }
+    const { error } = await supabaseClient.rpc("end_sync_task", {
+      s_target: context.spaceId,
+      s_function: "embedding",
+      s_worker: worker,
+      s_status: status,
+    });
+    if (error) {
+      console.error("endSyncTask: Error calling end_sync_task:", error);
+    }
+  } catch (error) {
+    console.error("endSyncTask: Error calling end_sync_task:", error);
+  }
+};
+
+export const proposeSyncTask = async (): Promise<SyncTaskInfo> => {
+  try {
+    const supabaseClient = await getLoggedInClient();
+    const context = await getSupabaseContext();
+    if (!context) {
+      console.error("proposeSyncTask: Unable to obtain Supabase context.");
+      return {
+        lastUpdateTime: null,
+        spaceId: 0,
+        worker: "",
+        shouldProceed: false,
+      };
+    }
+    const worker = window.roamAlphaAPI.user.uid();
```

Comment:
This also throws errors but also works

Reply:
I think you mean shows linting error? ("Throwing" suggests something else!) But yes, there is a mismatch here.